corehq/apps/domain/tests/test_api_key_auth.py

```diff
@@ -36,3 +39,48 @@ def test_credentials_in_POST():
     request = RequestFactory().post('/myapi/', {'username': USERNAME, 'api_key': API_KEY})
     assert has_credentials(request)
     assert not has_credentials(request, allow_creds_in_data=False)
+
+
+class AuthenticationTestBase(TestCase):
+    domain = 'api-key-test'
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.factory = RequestFactory()
+        cls.domain_obj = create_domain(name=cls.domain)
```

Comment:
```suggestion
        cls.domain_obj = create_domain(name=cls.domain)
        cls.addClassCleanup(cls.domain_obj.delete)
```
This ensures the domain object will still be deleted if a crash is encountered during test.

Reply:
Can you elaborate on that?  I thought this was essentially the same as calling this in `tearDownClass`.  Do you mean that if there's an error in `setUpClass` this will get called, while `tearDownClass` wouldn't?  Looks like that is indeed the case, thanks for pointing it out: https://docs.python.org/3/library/unittest.html#unittest.TestCase.addClassCleanup

Reply:
>  Do you mean that if there's an error in `setUpClass` this will get called, while `tearDownClass` wouldn't?

Exactly.